### DISCUSSION TITLE
Automatic differentiation support

### DISCUSSION THREAD

**Discussion by papachristoumarios** [GeomScale/volesti#203] [Ideas]
It would be a very nice feature to include automatic differentiation support within the functors, so that the gradients can be computed from the automatic differentiation software.

Automatic differentiation references and libraries: 

1. https://mc-stan.org/users/interfaces/math.html
2. https://github.com/autodiff/autodiff
3. https://www.boost.org/doc/libs/master/libs/math/doc/html/math_toolkit/autodiff.html

**Comment by vissarion** [comment]
Cool idea!

We have to check the possibilities. From your list above (3) is the simpler to use since `volesti` depends on boost already. 

So the question is: is `boost`'s `autodiff` functionality sufficient for volesti needs? What are the features needed? 

Here is a discussion comparing autodiff in `stan` and `boost` https://discourse.mc-stan.org/t/autodiff-in-boost/7440/11
